projection/ibc_channel_message/ibc_channel_message.go

```diff
@@ -0,0 +1,316 @@
+				Success:         true,
+				Error:           "",
+				Sender:          "",
+				Receiver:        "",
+				Denom:           "",
+				Amount:          "",
+				MessageType:     typedEvent.MsgName,
+				Message:         typedEvent,
+			}
+			messages = append(messages, message)
+
+		} else if typedEvent, ok := event.(*event_usecase.MsgIBCChannelOpenConfirm); ok {
+
+			channelID := typedEvent.Params.ChannelID
+
+			message := view.IBCChannelMessageRow{
+				ChannelID:       channelID,
+				BlockHeight:     height,
+				BlockTime:       blockTime,
+				TransactionHash: typedEvent.TxHash(),
+				Success:         true,
+				Error:           "",
+				Sender:          "",
+				Receiver:        "",
+				Denom:           "",
+				Amount:          "",
+				MessageType:     typedEvent.MsgName,
+				Message:         typedEvent,
+			}
+			messages = append(messages, message)
+
+		} else if typedEvent, ok := event.(*event_usecase.MsgIBCTransferTransfer); ok {
+
+			channelID := typedEvent.Params.SourceChannel
+
+			message := view.IBCChannelMessageRow{
+				ChannelID:       channelID,
+				BlockHeight:     height,
+				BlockTime:       blockTime,
+				TransactionHash: typedEvent.TxHash(),
+				Success:         true,
+				Error:           "",
+				Sender:          typedEvent.Params.Sender,
+				Receiver:        typedEvent.Params.Receiver,
+				Denom:           typedEvent.Params.Token.Denom,
+				Amount:          strconv.FormatUint(typedEvent.Params.Token.Amount, 10),
+				MessageType:     typedEvent.MsgName,
+				Message:         typedEvent,
+			}
+			messages = append(messages, message)
+
+		} else if typedEvent, ok := event.(*event_usecase.MsgIBCRecvPacket); ok {
+
+			channelID := typedEvent.Params.Packet.DestinationChannel
+
+			message := view.IBCChannelMessageRow{
+				ChannelID:       channelID,
+				BlockHeight:     height,
+				BlockTime:       blockTime,
+				TransactionHash: typedEvent.TxHash(),
+				Success:         typedEvent.Params.MaybeFungibleTokenPacketData.Success,
+				Sender:          typedEvent.Params.MaybeFungibleTokenPacketData.Sender,
+				Receiver:        typedEvent.Params.MaybeFungibleTokenPacketData.Receiver,
+				Denom:           typedEvent.Params.MaybeFungibleTokenPacketData.Denom,
+				Amount:          typedEvent.Params.MaybeFungibleTokenPacketData.Amount.String(),
+				MessageType:     typedEvent.MsgName,
+				Message:         typedEvent,
+			}
+			if typedEvent.Params.PacketAck.MaybeError != nil {
+				message.Error = *typedEvent.Params.PacketAck.MaybeError
+			}
+
+			messages = append(messages, message)
+
+		} else if typedEvent, ok := event.(*event_usecase.MsgIBCAcknowledgement); ok {
+
+			channelID := typedEvent.Params.Packet.SourceChannel
+
+			message := view.IBCChannelMessageRow{
+				ChannelID:       channelID,
+				BlockHeight:     height,
+				BlockTime:       blockTime,
+				TransactionHash: typedEvent.TxHash(),
+				Success:         typedEvent.Params.MaybeFungibleTokenPacketData.Success,
```

Comment:
This may break if the packet is not fungible token transfer

Reply:
Good catch! Let me change it.